Content.Server/_Goobstation/Antag/LastRolledAntagManager.cs

```diff
@@ -0,0 +1,106 @@
+// SPDX-FileCopyrightText: 2025 GoobBot <uristmchands@proton.me>
+// SPDX-FileCopyrightText: 2025 Ilya246 <ilyukarno@gmail.com>
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+using Content.Server.Database;
+using Robust.Server.Player;
+using Robust.Shared.Asynchronous;
+using Robust.Shared.Network;
+using System;
+using System.Threading.Tasks;
+
+// has to be in Content.Server to exist
+namespace Content.Server._Goobstation.Antag
+{
+    public sealed class LastRolledAntagManager
+    {
+        [Dependency] private readonly IServerDbManager _db = default!;
+        [Dependency] private readonly ITaskManager _task = default!;
+        [Dependency] private readonly IPlayerManager _player = default!;
+        private readonly List<Task> _pendingSaveTasks = new();
+        private ISawmill _sawmill = default!;
+
+        public void Initialize()
+        {
+            _sawmill = Logger.GetSawmill("last_antag");
+        }
+
+        /// <summary>
+        /// Saves last rolled values to the database before allowing the server to shutdown.
+        /// </summary>
+        public void Shutdown()
+        {
+            _task.BlockWaitOnTask(Task.WhenAll(_pendingSaveTasks));
+        }
+
+        /// <summary>
+        /// Sets a player's last rolled antag time.
+        /// </summary>
+        public TimeSpan SetLastRolled(NetUserId userId, TimeSpan to)
+        {
+            var oldTime = Task.Run(() => SetTimeAsync(userId, to)).GetAwaiter().GetResult();
+            _sawmill.Info($"Setting {userId} last rolled antag to {to} from {oldTime}");
+            return oldTime;
+        }
+
+        /// <summary>
+        /// Gets a player's last rolled antag time.
+        /// </summary>
+        public TimeSpan GetLastRolled(NetUserId userId)
+        {
+            return Task.Run(() => GetTimeAsync(userId)).GetAwaiter().GetResult();
+        }
+
+        #region Internal/Async tasks
+
+        /// <summary>
+        /// Sets a player's last rolled antag time.
+        /// </summary>
+        private async Task SetTimeAsyncInternal(NetUserId userId, TimeSpan time, TimeSpan oldTime)
+        {
+            Task<bool> setTimeTask = _db.SetLastRolledAntag(userId, time, cancellationToken);
```

Comment:
```suggestion
            Task<bool> setTimeTask = _db.SetLastRolledAntag(userId, time);
```
whoops i forgot you were a chud that doesn't use cancellation tokens